src/physics/linear_filament.rs

```diff
@@ -138,3 +145,504 @@ pub fn inductance_piecewise_linear_filaments(
+
+        for j in 0..m {
+            // Get distance from middle of the filament segment to the observation point
+            let rx = xp[j] - xmid; // [m]
+            let ry = yp[j] - ymid; // [m]
+            let rz = zp[j] - zmid; // [m]
+            let rnorm = rss3(rx, ry, rz); // [m]
+
+            let c = current / rnorm;
+
+            ax[j] = c.mul_add(dlxi, ax[j]); // [V-s/m]
+            ay[j] = c.mul_add(dlyi, ay[j]);
+            az[j] = c.mul_add(dlzi, az[j])
+        }
+    }
+
+    for j in 0..m {
+        ax[j] *= MU0_OVER_4PI;
+        ay[j] *= MU0_OVER_4PI;
+        az[j] *= MU0_OVER_4PI;
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use std::f64::consts::PI;
+
+    use super::*;
+
+    /// Div/0-resistant approximate comparison
+    fn approx(truth: f64, val: f64, rtol: f64, atol: f64) -> bool {
+        let abs_err = (val - truth).abs();
+        let lim = rtol * truth.abs() + atol;
+        abs_err < lim
+    }
+
+    /// Check that B = curl(A)
+    #[test]
+    fn test_vector_potential() {
+        // One super basic filament as the source
+        let xyz = [0.0];
+        let dlxyz = [1.0];
+
+        // Build a second scattering of filament locations as the target
+        const NFIL: usize = 10;
+        let xfil2: Vec<f64> = (0..NFIL).map(|i| (i as f64).sin() + PI).collect();
+        let yfil2: Vec<f64> = (0..NFIL).map(|i| (i as f64).cos() - PI).collect();
+        let zfil2: Vec<f64> = (0..NFIL)
+            .map(|i| (i as f64) - (NFIL as f64) / 2.0 + PI)
+            .collect();
+        let xyzfil2 = (
+            &xfil2[..=NFIL - 2],
+            &yfil2[..=NFIL - 2],
+            &zfil2[..=NFIL - 2],
+        );
+
+        let dlxfil2: Vec<f64> = (0..=NFIL - 2).map(|i| xfil2[i + 1] - xfil2[i]).collect();
+        let dlyfil2: Vec<f64> = (0..=NFIL - 2).map(|i| yfil2[i + 1] - yfil2[i]).collect();
+        let dlzfil2: Vec<f64> = (0..=NFIL - 2).map(|i| zfil2[i + 1] - zfil2[i]).collect();
+        let dlxyzfil2 = (&dlxfil2[..], &dlyfil2[..], &dlzfil2[..]);
+
+        let xmid2: Vec<f64> = xfil2
+            .iter()
+            .zip(dlxfil2.iter())
+            .map(|(x, dx)| x + dx / 2.0)
+            .collect();
+        let ymid2: Vec<f64> = yfil2
+            .iter()
+            .zip(dlyfil2.iter())
+            .map(|(x, dx)| x + dx / 2.0)
+            .collect();
+        let zmid2: Vec<f64> = zfil2
+            .iter()
+            .zip(dlzfil2.iter())
+            .map(|(x, dx)| x + dx / 2.0)
+            .collect();
+
+        // Check against Neumann's formula for mutual inductance
+        let outx = &mut [0.0; NFIL - 1];
+        let outy = &mut [0.0; NFIL - 1];
+        let outz = &mut [0.0; NFIL - 1];
+        vector_potential_linear_filament(
+            (&xmid2, &ymid2, &zmid2),
+            (&xyz, &xyz, &xyz),
+            (&dlxyz, &dlxyz, &dlxyz),
+            &[1.0],
+            (outx, outy, outz),
+        )
+        .unwrap();
+        let a_dot_dl: Vec<f64> = (0..NFIL - 1)
```

Comment:
I think some comments explaining the math are needed to make this test legible to future readers.
Perhaps this?
```suggestion
        // Here the mutual inductance of the two filaments is calculated from the
        // vector potential at filament 2 due to 1 ampere of current flowing in filament 1.
        // By Stokes' theorem, the line integral of A over filament 2 is equal to the
        // magnetic flux through a surface bounded by filament 2. The flux through
        // filament 2 due to 1 ampere of current in filament 1 is the mutual inductance.
        // (We are stretching the applicability of Stokes' therorem because the filaments
        // are not closed loops)
        let a_dot_dl: Vec<f64> = (0..NFIL - 1)
```